modules/server/src/network/utils/sqon.ts

```diff
@@ -0,0 +1,20 @@
+import SQONBuilder, { SQON } from '@overture-stack/sqon-builder';
+import { failure, Result, success } from '../result';
+
+/**
+ * Attempts to convert a variable to a SQON. A Result is returned, where if successful
+ * the type checked SQON object is returned, and if a failure occurs then the error data
+ * from the SQONBuilder is returned as the failure message.
+ *
+ * @param filter
+ * @returns
+ */
+export const convertToSqon = (filter: unknown): Result<SQON, 'INVALID_SQON', void> => {
+	try {
+		const output = SQONBuilder.from(filter);
+		return success(output);
+	} catch (error: unknown) {
+		const message = `${error}`; // Will convert the error to a string, works fine with the expected ZodError without reuiring importing Zod to arranger
```

Comment:
Where is Zod being implemented that would provide a ZodError? SQON-Builder? 🤔 if so, sounds like something that should be normalised as a string within that library, to prevent needing this hack everywhere that doesn't need/use Zod.

```suggestion
		const message = `${error}`; // Will convert the error to a string, works fine with the expected ZodError without requiring importing Zod to Arranger
```

Reply:
Yes, from SQONBuilder. Can possibly return a friendly error message too, right now this is full stack trace Error from SQONBuilder.